File: naked-generator/BMSinternal/py/intra_net_bms_sync.py
from enum import Enum
from struct import pack, unpack
from collections import namedtuple


MASTER_SYNC = namedtuple('MASTER_SYNC', 'time')
MASTER_SYNC_schema = '<I'

CELLBOARD_TOKEN = namedtuple('CELLBOARD_TOKEN', 'sender_cellboard time')
CELLBOARD_TOKEN_schema = '<BI'


def serialize_MASTER_SYNC(struct: MASTER_SYNC) -> bytearray:
	return pack(MASTER_SYNC, *tuple(struct))


def serialize_CELLBOARD_TOKEN(struct: CELLBOARD_TOKEN) -> bytearray:
	return pack(CELLBOARD_TOKEN, *tuple(struct))


def deserialize_MASTER_SYNC(buffer: bytearray) -> MASTER_SYNC:
	return MASTER_SYNC._make(unpack(MASTER_SYNC_schema, buffer))


def deserialize_CELLBOARD_TOKEN(buffer: bytearray) -> CELLBOARD_TOKEN:
	return CELLBOARD_TOKEN._make(unpack(CELLBOARD_TOKEN_schema, buffer))

File: naked-generator/BMSinternal/c/intra_net_bms_sync.c
#include <string.h>
#include <assert.h>
#include <stdio.h>

#include "intra_net_bms_sync.h"

void serialize_MASTER_SYNC(MASTER_SYNC* master_sync, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(MASTER_SYNC));
	memcpy(buffer, master_sync, sizeof(STEER_STATUS));
}
void serialize_CELLBOARD_TOKEN(CELLBOARD_TOKEN* cellboard_token, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(CELLBOARD_TOKEN));
	memcpy(buffer, cellboard_token, sizeof(STEER_STATUS));
}

void deserialize_MASTER_SYNC(uint8_t* buffer, size_t buf_len, MASTER_SYNC* master_sync) {
	assert(buf_len >= sizeof(MASTER_SYNC));
	memcpy(master_sync, buffer, sizeof(STEER_STATUS));
}
void deserialize_CELLBOARD_TOKEN(uint8_t* buffer, size_t buf_len, CELLBOARD_TOKEN* cellboard_token) {
	assert(buf_len >= sizeof(CELLBOARD_TOKEN));
	memcpy(cellboard_token, buffer, sizeof(STEER_STATUS));
}
